### ISSUE TITLE
question about statefulness

### ISSUE THREAD

**Issue by kathyfan** [flomlo/ntm_keras#8]
Apologies - it seems like the NTM just needs to be trained on more epochs than some of the other models I was using. This is no longer an issue. 

I did have another question about the "stateful=True" for the controller though -- is this necessary? And could you explain your comment that "Stateful models do not work anymore"? Thank you!

**Comment by MortezaKarimian** [comment]
by googling about your question we can understand that:

@_All the RNN or LSTM models are stateful in theory. These models are meant to remember the entire sequence for prediction or classification tasks. However, in practice, you need to create a batch to train a model with backprogation algorithm, and the gradient can't backpropagate between batches. This means that if you have a long time series which does not fit into a single batch, you need to divide the time series into multiple sub-time series and each sub time series goes to separate batch. Then LSTM only remember what happened within a batch. At the initial time point of every batch, states are initialized and set to 0. No previous information. This is very unfortunate because RNN or LSTM are introduced to remember all the past history to predict the next time point. Nope, it only remembers what happened within the batch time series length!_@ ref: [link](https://fairyonice.github.io/Stateful-LSTM-model-training-in-Keras.html)

and also:

- stateful is useful if you wish to save the state of the neurons for the next training session instead of resetting it
- In stateless mode, long term memory does not mean that the LSTM will remember the content of the previous batches
- stateful=True means that you keep the final state for every batch and pass it as initial state for the next batch
